### ISSUE TITLE
OIDCAuthenticator - makes sense?

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/jupyterhub#2462]
Hey, I figure it would make sense to create a Authenticator that can handle a OpenID Connect endpoints like ".well-known/openid-configuration" providing configuration information, like here: 
![image](https://user-images.githubusercontent.com/3837114/54011858-4f107080-4174-11e9-8433-5ab3b8de8339.png)

@minrk does it make sense to you? This is something I'd be happy to invest time in, one way or another. I have spent sooooo much time working with auth lately and this is one of the challenging things to setup in general I think, but OIDC implemented OK can make life a bit easier.

**Comment by betatim** [comment]
It could be interesting to extend https://github.com/jupyterhub/oauthenticator/blob/master/oauthenticator/generic.py so that it can be configured with just the "well known" URL and get all other links from there.

The generic OAuthenthicator works very well with Auth0 and friends. As far as I can tell OIDC is mostly just the complicated name used in companies for OAuth2.

**Comment by consideRatio** [comment]
I think OIDC is a standard built on top of OAuth2 standardized things, it adds details about how a userinfo endpoint should work for example I believe, and how you should also have this configuration endpoint, often accessible at https://example.com/.../.well-known/openid-configuration. And this endpoint should adhere to various rules etc.

**Comment by manics** [comment]
@consideRatio You're correct, OIDC is basically OAuth2 with a few extras, though it's surprising how badly documented that is (took me over a week to realise!). I've just written a very experimental OAuth2/OIDC app for one of our apps, so if it's helpful I can point you to the differences between OAuth2 and OIDC, and hopefully save you some of the pain I went through 😀.

**Comment by consideRatio** [comment]
;D haha yes @manics that is what I experienced as well, but I must say I have spent a lot more than a week by now and still is vague about things ^^

**Comment by consideRatio** [comment]
Related: https://github.com/jupyterhub/oauthenticator/issues/245

**Comment by Zsailer** [comment]
@consideRatio and @manics For reference, I've found [this online book](https://www.oauth.com/) by the Okta team *extremely* helpful for understanding the relationship between OAuth and OIDC.
